org.scala-ide.sdt.core.tests/src/org/scalaide/core/compiler/NamePrinterTest.scala

```diff
@@ -387,15 +387,142 @@ class NamePrinterTest {
+         |class WithSimpleAnnotation
+      """,
+      "SimpleAnnotation")
+  }
+
+  @Test
+  def testWithAnnotationOnDefOnParam() {
+    testWith(
+      """|import scala.annotation._
+         |class AnnotationOnDefOnParam(val name: String) extends StaticAnnotation
+         |object AnnotationOnDefOnParam {
+         |  @AnnotationOnDefOnParam(name/**/ = "Hallo")
+         |  def test = 32
+         |}""",
+      "AnnotationOnDefOnParam(name: String)")
+  }
+
+  @Test
+  def testWithAnnotationOnDef() {
+    testWith(
+      """|import scala.annotation._
+         |class AnnotationOnDef(val name: String) extends StaticAnnotation
+         |object AnnotationOnDef {
+         |  @AnnotationOnDef/**/(name = "Hallo")
+         |  def test = 32
+         |}""",
+      "AnnotationOnDef")
+  }
+
+  @Test
+  def testWithJavaxGeneratedAnnotationOnClass() {
+    testWith(
+      """|import javax.annotation._
+         |@Generated/**/(Array("today"))
+         |class WithJavaxGeneratedAnnotationOnClass""",
+      "javax.annotation.Generated")
+  }
+
+  @Test
+  def testWithJavaxGeneratedAnnotationOnMethod() {
+    testWith(
+      """|import javax.annotation._
+         |class WithJavaxGeneratedAnnotationOnMethod {
+         |  @Generated/**/(Array("yesterday"))
+         |  def method = Unit
+         |}""",
+      "javax.annotation.Generated")
+  }
+
+  @Test
+  def testWithJavaxResourceAnnotationOnMethod() {
+    testWith(
+      """|import javax.annotation._
+         |class WithJavaxResourceAnnotationOnResource {
+         |  @Resource/**/(name = "test", lookup = "anywhere")
+         |  def method = Unit
+         |}""",
+      "javax.annotation.Resource")
+  }
+
+  @Test
+  def testWithAuxillaryCtor() {
+    testWith(
+      """|class WithAxillaryCtor(str: String) {
+         |  def this(i: Int) = this(i.toString)
+         |  def foo() { new WithAxillaryCtor(/**/33) }
+         |}""",
+      "WithAxillaryCtor(i: Int)")
+  }
+
+  @Test
+  def testWithMethodCallOnGeneric() {
+    testWith(
+      """|class WithMethodCallOnGeneric[A](a: A) {
+         |  def method[B, C](b: B, c: C) = b.hashCode + c.hashCode
+         |  def foo = {
+         |    val tst = new WithMethodCallOnGeneric("string")
+         |    tst.method(/**/1L, 0.0)
+         |  }
+         |}""",
+      "WithMethodCallOnGeneric.method[B, C](b: B, c: C)")
+  }
+
   private def testQnameWith(input: String, expected: Option[String]) {
     val source = input.stripMargin
-    val cu = mkScalaCompilationUnit(source)
+    val cu = prepareCompilationUnit(source)
     val offset = verifyOffset(source.indexOf("/**/") - 1)
     val namePrinter = new NamePrinter(cu)
     val res = namePrinter.qualifiedNameAt(offset)
     assertEquals(expected, res)
   }
 
+  private def prepareCompilationUnit(source: String) = {
+    waitForReconcile(mkScalaCompilationUnit(source, true))
+  }
+
+  private def waitForReconcile(cu: ScalaCompilationUnit) = {
+    cu.initialReconcile().get match {
```

Comment:
I just added that after getting test failures hoping that it would make a difference; since that does not seem to be the case I can as well remove it...